pkg/deviceclaimingserver/enddevices/enddevices.go

```diff
@@ -43,6 +44,15 @@ type EndDeviceClaimer interface {
 	GetClaimStatus(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers) (*ttnpb.GetClaimStatusResponse, error)
 	// Unclaim releases the claim on an End Device.
 	Unclaim(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers) (err error)
+
+	// BatchUnclaim releases the claim on a batch of end devices.
+	// All devices in a batch must have the same Join EUI.
+	// If the returned error is of type `InvalidArgument`,
+	// the returned error map will contain errors for individual devices.
```

Comment:
This is not idiomatic go; if `err != nil` then the other return values are undefined.

You should simply return error here, and declare an error type that implements the error interface and includes the map, and check with `errors.Is` or `errors.As` what error it is and get to the details.